src/subdomains/generic/user/models/auth/auth.service.ts

```diff
@@ -137,7 +137,7 @@ export class AuthService {
     const userData = userDataId && (await this.userDataService.getUserData(userDataId, { users: true }));
     const primaryUser = userId && (await this.userService.getUser(userId));
 
-    const custodyProvider = await this.custodyProviderService.getWithMasterKey(dto.signature);
+    const custodyProvider = !dto.key ? await this.custodyProviderService.getWithMasterKey(dto.signature) : undefined;
```

Comment:
Maybe just `await this.custodyProviderService.getWithMasterKey(dto.signature).catch(() => undefined);`?